### PR TITLE
More icon interface

### PR DESCRIPTION
**Pull request by Befkadu1**
fix https://github.com/Lundalogik/lime-elements/issues/2686
## Review:
- [ ] Commits are [atomic](https://seesparkbox.com/foundry/atomic_commits_with_git)
- [ ] Commits have the correct *type* for the changes made
- [ ] Commits with *breaking changes* are marked as such

### Browsers tested:
(Check any that applies, it's ok to leave boxes unchecked if testing something didn't seem relevant.)

Windows:
- [ ] Chrome
- [ ] Edge
- [ ] Firefox

Linux:
- [ ] Chrome
- [ ] Firefox

macOS:
- [ ] Chrome
- [ ] Firefox
- [ ] Safari

Mobile:
- [ ] Chrome on Android
- [ ] iOS

### CODE REVIEW

**Review by Kiarokh — APPROVED** [review]

File: src/components/menu-list/menu-list-renderer.tsx [hunk 1]
```diff
@@ -139,7 +139,7 @@ export class MenuListRenderer {
                 data-index={index}
                 {...attributes}
             >
-                {item.icon ? this.renderIcon(this.config, item) : null}
+                {this.renderIcon(this.config, item)}
```
Comment (adrianschmidt) [inline src/components/menu-list/menu-list-renderer.tsx:142]: 😬

This change makes the list crash if any item doesn't have `icon` specified 😞

Reports of several customers having this problem: https://forum.lime-crm.com/t/query-visualizer-not-working-with-newer-lime-elements/5723

The problem is that `renderIcon` uses `getIconName`, which in turn does:

```ts
if (typeof icon === 'object' && 'name' in icon) {
    return icon.name;
}
```

The problem is that `typeof null` returns `'object'`, so this tries to run `'name' in null`, which causes an error.

I think `get-icon-props.ts` is a pretty perfect example of how helper-functions can be broken out into a separate file, to make them testable. But unfortunately, there are no tests for that file 🙈 😄

I'll look into adding some tests and fixing the bug.

@Befkadu1 @Kiarokh

Comment (Kiarokh) [reply src/components/menu-list/menu-list-renderer.tsx:142]: 🙏

**Pull request merged**